llmfoundry/models/hf/hf_causal_lm.py

```diff
@@ -193,6 +194,8 @@ def build_inner_model(
         config_overrides: Dict[str, Any],
         load_in_8bit: bool,
         pretrained: bool,
+        config_fn: Optional[Callable] = AutoConfig,
```

Comment:
do you really intend for these to be functions? They should be classes that support `.from_pretrained`/`.from_config`, which return model/config as appropriate, right?

Reply:
true - but was not sure what "type" they are since these are classes?

Reply:
ok - I think I've fixed this.

Reply:
Yeah, I mean, the most complete way would be to define the interface that is require to be implemented. I don't think that is necessary. Instead, how about we just specify it as the type `Type` (its not `Optional`, because it can't be `None`). And then in code, we can check `hasattr(auto_config_class, 'from_pretrained')` etc and raise error if not defined. and can also raise errors if they return the wrong thing.

Reply:
The correct type is actually this private class from transformers ithink https://github.com/huggingface/transformers/blob/9d2ab8824c2a820a0ac9f2cd884ef838ca77b481/src/transformers/models/auto/auto_factory.py#L403

Reply:
and autoconfig has no base class (https://github.com/huggingface/transformers/blob/9d2ab8824c2a820a0ac9f2cd884ef838ca77b481/src/transformers/models/auto/configuration_auto.py#L859)